Change Measurement to Property where appropriate (#226)
ietf-rats-wg/draft-ietf-rats-corim: draft-ietf-rats-corim.md

# Concise Module Identifier (CoMID) {#sec-comid}

A CoMID tag contains information about subjects such as hardware, firmware, or module composition.

Each CoMID has a unique ID that is used to unambigously identify CoMID instances when cross referencing CoMID tags, for example in typed link relations, or in a CoBOM tag.

> A CoMID represents expectations for Evidence appraisal with "triples".

Reviewer: Is the use of `expectations` trying to combine the meanings of reference values and endorsed values?

{::include cddl/group-id-type-choice.cddl}
~~~

##### Properties

> A Property is a key-value pair that in CoRIM represents a single piece of actual state or a possible state.

Reviewer: Is now unfortunately defined in a list of terms later in the document. Maybe:

```suggestion
A Property in CoRIM represents a single piece of actual state or a possible state (see {{sec-verifier-abstraction}} below).
```

A Property in CoRIM always appears within a triple.
The triple expresses the intended meaning of the Property.
In all cases, a Property is made by an authority, such as an Attester, Endorser, or Reference Value Provider.

> A Property from an Attester is called Evidence.

Reviewer: ```suggestion
A Property from an Attester is conveyed via Evidence.
```

A Property in a `reference-triple-record` is called a Reference Value.

> A Property in another triple is called an Endorsement.

Reviewer: Does the list in [5.1.4 Triples](https://ietf-rats-wg.github.io/draft-ietf-rats-corim/draft-ietf-rats-corim.html#section-5.1.4) fit that statement?

A Reference Value triple is expressible as a conditional Endorsement that if the Attester made some Properties, then the RVP also makes those Properties.

> A Property may be called an assertion, property, measurement, or many other classifications based on its role in Evidence appraisal.

Reviewer: I assume that property and claim have to switch places here.

```suggestion
A Property may be called an assertion, claim, measurement, or many other classifications based on its role in Evidence appraisal.
```
And then, why do we say that here in the first place? Should that go to the definition of Property?

The supply chain entity that is responsible for providing the properties is by default the CoRIM signer.

If a different entity is authorized to provide property values, the `authorized-by` statement can be supplied in the `properties-map`.
<!-- end suggestion -->

~~~ cddl
{::include cddl/properties-map.cddl}
~~~

The following describes each member of the `properties-map`:

* `pkey` (index 0): An optional unique identifier of the property-associated
  (sub-)environment.  See {{sec-comid-pkey}}.

* `pval` (index 1): The properties associated with the (sub-)environment.
  Described in {{sec-comid-pval}}.

* `authorized-by` (index 2): The cryptographic identity of the individual or organization that is
 the designated authority for this property. For example, producer of the property or a delegated supplier.

###### Property Keys {#sec-comid-pkey}

The types defined for a property key are OID, UUID or uint.

~~~ cddl
{::include cddl/property-key-type-choice.cddl}
~~~

###### Property Values {#sec-comid-pval}

A `property-values-map` contains properties associated with a certain
environment. Depending on the context (triple) in which they are found,

> elements in a `property-values-map` can represent class or instance

Reviewer: ```suggestion
members of a `property-values-map` can represent class or instance
```

> properties. Note that some of the elements have instance scope only.

Reviewer: ```suggestion
properties. Note that some of the member types have instance scope only.
```

{::include cddl/group-id-type-choice.cddl}
~~~

##### Claims

> A Claim is a key, value pair that in CoRIM represents a single piece of actual state or a reference state.

Reviewer 2: ["key-value pair"](https://en.wikipedia.org/wiki/Name–value_pair)

Reviewer 3: If we're following the endorsements draft terminology it would read something like: "A Claim is a key-value pair that in CoRIM represents a single piece of actual or possible state.

{::include cddl/group-id-type-choice.cddl}
~~~

##### Claims

> A Claim may either be a measurement that can be directly compared with attestation evidence, or it can be an endorsement of a quality that is not directly comparable with evidence.

Reviewer 2: Trying to recover a suggestion that got lost in the discussion.
```suggestion
A Claim represents a single piece of actual or reference state.
A Claim may either be a measurement that can be directly compared with attestation evidence, or it can be an endorsement of a quality that is not directly comparable with evidence.
```
There was a further suggestion to s/measurement/reference value/ in L848

A Claim in CoRIM always appears within a triple.
The triple expresses the intended meaning of the Claim.
In all cases, a Claim is made by an authority, such as an Attester, Endorser, or Reference Value Provider.

A Claim from an Attester is called Evidence.
A Claim in a `reference-triple-record` is called a Reference Value.
A Claim in another triple is called an Endorsement.
A Reference Value triple is expressible as a conditional Endorsement that if the Attester made some Claims, then the RVP also makes those Claims.

A Claim may be called an assertion, property, measurement, or many other classifications based on its role in Evidence appraisal.
Attester-authenticated Claims are typically called measurements.
Measurements can be of a variety of components including software, firmware, configuration files, read-only memory, fuses, IO ring configuration, partial reconfiguration regions, etc.
Measurements comprise raw values, digests, or status information.

There may be multiple Claims made of the same Target Environment.
A single Claim may describe a combination of multiple components.

> The scope of a Claim depends on the specificity of the environment in the triple it appears in.

Reviewer 2: ```suggestion
The triple’s environment defines the scope of a Claim.
```

The following terms are used:

{: vspace="0"}
Property:

> : A piece of information, in the form of a key-value pair.

Reviewer: Here the Property and the Claim definition that RFC9334 inherits collide a bit. Maybe something like:

```suggestion
: A piece of information, in the form of a Claim as specified in {{-rats-arch}} where the name of the name/value pair is always an integer with an annotated text string name.
```

Reviewer 2: ```
: Compare: Claim {{-rats-arch}}
```
